unowmooc/edx-platform
# Allow each instance to configure restrictions on allowed email addresses

OC-1200 (see also previous task OC-318)

This adds a new configuration option to edX platform that can limit registration to specific email patterns (regex).

For complete application of this feature, one must configure the whitelist of accepted email addresses in both `~/lms.env.json` and `~/cms.env.json`.

**Test Instructions**
1. Test on a [**Cypress** devstack](https://github.com/edx/configuration/wiki/edX-Developer-Stack) (I'll open an upstream `master` PR once this is reviewed here)
2. In `~/lms.env.json` and `~/cms.env.json`, add something like:  
   
   ``` json
   "REGISTRATION_EMAIL_PATTERNS_ALLOWED": [".*@opencraft\\.com"],
   ```
3. Now, try registering opencraft and non-opencraft emails on both the LMS (with combined registration enabled and disabled) and the CMS. Only emails that match the regex should be allowed.
